lwotg/flows.go

```diff
@@ -0,0 +1,80 @@
+package lwotg
+
+import (
+	"github.com/open-traffic-generator/snappi/gosnappi/otg"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"k8s.io/klog"
+)
+
+// TXRXFn is a function that handles the send and receive of packets for a particular flow.
+// When called it should begin tranmitting packets using the tx FlowController channels, and
+// receiving them using the rx FlowController. Tx and Rx should be spawned into new goroutines
+// and the function should return. Telemetry that is published by the flow should be written
+// to the corresponding FlowController GNMI channel. Encountered errors should be written to
+// the error channel.
+type TXRXFn func(tx, rx *FlowController)
+
+// FlowController acts as the control mechanism for a specific direction of a specific flow.
+type FlowController struct {
+	// Stop is a channel used to indicate that the function that has been called should
+	// cease to send or receive packets.
+	Stop chan struct{}
+}
+
+// FlowGeneratorFn is a function that takes an input OTG Flow and, if it is able to, returns
+// a TXRXFn that controls that flow. If the function is not able to handle the flow (e.g., it
+// is an IP flow and the function can handle only MPLS) it returns no TXRXFn, and sets the
+// bool return argument to false. If the function can handle the flow, it returns a TXRXFn that
+// can be used to start the flow, with the bool return argument set to true. Errors encountered
+// whilst generating a TXRXFn for a flow that the function can return should be returned and
+// are treated as errors that should be returned to the caller.
```

Comment:
what's the intended behavior if multiple functions are able to handle the flow? is that a possible scenario? right now the code just picks the first one that can handle the flow, and returns.

Reply:
It is definitely a possible scenario. :-)

My intention was as you say, to ensure that the first handler is the one that handles the flow. In general, this probably works because we have no "default" flow handlers. In the future, one might want to improve this to give some form of registered order if we added default ones.

For now, I propose to add something to the documentation string here, and a TODO that mentions that we might want this to be improved in the future.